dashboard/DashboardDB.py

```diff
@@ -0,0 +1,344 @@
+import os
+import shutil
+import unittest
+import sqlite3 as sql
+from collections import namedtuple
+from typing import Iterable, Union
+from datetime import datetime, date, timedelta
+
+import qcore.constants as const
+
+SQueueEntry = namedtuple(
+    "SQueue",
+    ["job_id", "username", "status", "name", "run_time", "est_run_time", "nodes", "cores"],
+)
+
+
+class DashboardDB:
+
+    date_format = "%d-%m-%Y"
+
+    def __init__(self, db_file: str):
+        """Opens an existing dashboard database file."""
+
+        if not os.path.isfile(db_file):
+            raise FileNotFoundError(
+                "Specified database {} does not exist. Use static create_db "
+                "function to create a new dashboard db file.".format(db_file)
+            )
+
+        self.db_file = db_file
+
+    @staticmethod
+    def get_daily_t_name(hpc: const.HPC):
+        return "{}_DAILY".format(hpc.value.upper())
+
+    @staticmethod
+    def get_squeue_t_name(hpc: const.HPC):
+        return "{}_SQUEUE".format(hpc.value.upper())
+
+    def update_daily_chours_usage(
+        self,
+        total_core_usage: float,
+        hpc: const.HPC,
+        day: Union[date, str] = date.today(),
+    ):
+        """Updates the daily core hours usage.
+
+        The core hour usage for a day is calculated as
+        last saved (for current day) TOTAL_CORE_USAGE - current total_core_usage.
+        """
+        # Do nothing if total_core_usage is None
+        if total_core_usage is None:
+            return
+
+        conn = sql.connect(self.db_file)
```

Comment:
Can we make a setup and teardown with and a 'with'? Seems like that would be the best practice.

Reply:
@jasonmotha  Can you show an example?

Reply:
have added this in commit   https://github.com/ucgmsim/slurm_gm_workflow/pull/68/commits/6c6224e377c429bcf175c4f0adeb242552af6a75